gdal/ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp

```diff
@@ -5156,7 +5163,7 @@ OGRErr GDALGeoPackageDataset::DeleteRasterLayer( const char* pszLayerName )
         sqlite3_free(pszSQL);
     }
 
-    if( eErr == OGRERR_NONE )
+    if( eErr == OGRERR_NONE && HasGriddedCoverageAncillaryTable() )
```

Comment:
this test should also be added for the deletion from gpkg_2d_gridded_tile_ancillary

Reply:
It could be better this way?

    if( eErr == OGRERR_NONE && HasGriddedCoverageAncillaryTable() )
    {
        char* pszSQL = sqlite3_mprintf(
                "DELETE FROM gpkg_2d_gridded_coverage_ancillary WHERE lower(tile_matrix_set_name) = lower('%q')",
                pszLayerName);
        eErr = SQLCommand(hDB, pszSQL);
        sqlite3_free(pszSQL);

        if( eErr == OGRERR_NONE )
        {
            pszSQL = sqlite3_mprintf(
                    "DELETE FROM gpkg_2d_gridded_tile_ancillary WHERE lower(tpudt_name) = lower('%q')",
                    pszLayerName);
            eErr = SQLCommand(hDB, pszSQL);
            sqlite3_free(pszSQL);
        }
    }